src/NAME/Elasticsearch/ElasticsearchVersionResolver.cs

```diff
@@ -0,0 +1,65 @@
+using NAME.Core;
+using NAME.Core.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace NAME.Elasticsearch
+{
+    internal class ElasticsearchVersionResolver : ConnectedVersionResolver
+    {
+        private IConnectionStringProvider connectionStringProvider;
+
+        public ElasticsearchVersionResolver(IConnectionStringProvider connectionStringProvider, int connectTimeout, int readWriteTimeout)
+            : base(connectTimeout, readWriteTimeout)
+        {
+            this.connectionStringProvider = connectionStringProvider;
+        }
+
+        public override async Task<IEnumerable<DependencyVersion>> GetVersions()
+        {
+            var versions = new List<DependencyVersion>();
+            var connectionString = string.Empty;
+
+            if (!this.connectionStringProvider.TryGetConnectionString(out connectionString))
+            {
+                throw new ConnectionStringNotFoundException(this.connectionStringProvider.ToString());
+            }
+
+            HttpWebResponse response;
+            try
+            {
+                var request = this.GetHttpWebRequest(connectionString, SupportedDependencies.Elasticsearch.ToString());
+                response = await request.GetResponseAsync() as HttpWebResponse;
+            }
+            catch (Exception e)
```

Comment:
You should catch the most relevant exception instead of the base `Exception`, in this case it is `System.Net.WebException`.